src/main/java/seedu/mark/logic/commands/AddFolderCommand.java

```diff
@@ -0,0 +1,76 @@
+package seedu.mark.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.mark.logic.commands.commandresult.CommandResult;
+import seedu.mark.logic.commands.exceptions.CommandException;
+import seedu.mark.logic.parser.CliSyntax;
+import seedu.mark.model.Model;
+import seedu.mark.model.bookmark.Folder;
+
+/**
+ * Creates a new folder in Mark.
+ */
+public class AddFolderCommand extends Command {
+
+    public static final String COMMAND_WORD = "folder";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Creates a new folder with the specified name "
+            + "Parameters: NAME (must not be an existing folder name) "
+            + CliSyntax.PREFIX_PARENT_FOLDER + "[PARENT_FOLDER]\n"
+            + "Example: " + COMMAND_WORD + " school ";
+    public static final String MESSAGE_DUPLICATE_FOLDER = "This folder already exists in Mark";
+    public static final String MESSAGE_PARENT_FOLDER_NOT_FOUND = "The parent folder %s doesn't exist in Mark";
+    public static final String MESSAGE_SUCCESS = "Folder %s added successfuly";
+
+    private final String folderName;
```

Comment:
Any reason the folder is stored as string instead of `Folder` object here? Cos from `Parser` to `Command`, the folder name gets converted from `String` to `Folder`, then here it gets converted back to the `String` then gets passed to Model to convert to `FolderStructure` again. Is it possible to use `Folder` object instead? So in the `FolderStructure` it can store a `Folder` and a list of `FolderStructure` as subfolders. Correct me if understand it wrongly :P

Reply:
ok makes sense, so we try not to have any String folderNames as much as possible. thanks!